tools/build_pytorch_libs.py

```diff
@@ -101,7 +101,10 @@ def run_cmake(version,
         get_cmake_command(),
         base_dir
     ]
-    if USE_NINJA:
+    generator = os.getenv('CMAKE_GENERATOR')
```

Comment:
I don't see how this will work. The script only knows how to build using a couple specific generators (the default per platform and ninja). Allowing a custom generator here would just cause the script to fail later. What is this line trying to accomplish? At the very least a comment would be useful.

Reply:
I checked the code in line 258-272. Yes, you are right. I tried to change this because the default generator of CMake under Windows is MSVC x86, which cannot be used to compile PyTorch. What do you think if I hardcode the cmake arg `-GVisual Studio 15 2017` if ninja is not used?

Reply:
Yeah, that is good!

Reply:
That also explains why I could never get it compile correctly on windows without using ninja. I want to get the build to a state where a clean checkout using `pytorch setup.py install` will work on windows as long as python/cmake are in the path and vsvarsall.bat has been run.